src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -52,6 +52,111 @@ public static FormulaValue Index_UO(IRContext irContext, FormulaValue[] args)
             }
         }
 
+        public static FormulaValue First_UO(IRContext irContext, UntypedObjectValue[] args)
+        {
+            var arg0 = (UntypedObjectValue)args[0];
+            var element = arg0.Impl;
+            var len = element.GetArrayLength();
+
+            if (len == 0)
+            {
+                return new BlankValue(irContext);
+            }
+
+            var result = element[0];
+
+            // Map null to blank
+            if (result == null || result.Type == FormulaType.Blank)
+            {
+                return new BlankValue(IRContext.NotInSource(FormulaType.Blank));
+            }
+
+            return new UntypedObjectValue(irContext, result);
+        }
+
+        public static FormulaValue Last_UO(IRContext irContext, UntypedObjectValue[] args)
+        {
+            var arg0 = (UntypedObjectValue)args[0];
+            var element = arg0.Impl;
+            var len = element.GetArrayLength();
+
+            if (len == 0)
+            {
+                return new BlankValue(irContext);
+            }
+
+            var result = element[len - 1];
+
+            // Map null to blank
+            if (result == null || result.Type == FormulaType.Blank)
+            {
+                return new BlankValue(IRContext.NotInSource(FormulaType.Blank));
+            }
+
+            return new UntypedObjectValue(irContext, result);
+        }
+
+        public static FormulaValue FirstN_UO(IRContext irContext, FormulaValue[] args)
+        {
+            var arg0 = (UntypedObjectValue)args[0];
+            var arg1 = (NumberValue)args[1];
+
+            var element = arg0.Impl;
+            var len = element.GetArrayLength();
+
+            if (len == 0)
+            {
+                return new BlankValue(irContext);
+            }
+
+            var list = new List<IUntypedObject>();
+            for (int i = 0; i < (int)arg1.Value && i < len; i++)
+            {
+                list.Add(element[i]);
+            }
+
+            var result = new ArrayUntypedObject(list);
+
+            // Map null to blank
+            if (result == null || result.Type == FormulaType.Blank)
```

Comment:
>result == null || result.Type == FormulaType.Blank [](http://example.com/codeflow?start=16&length=50)

This can never happen - result is an ArrayUntypedObject, and result.Type is ArrayType

Reply:
Fixed